paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/SparkSource.scala

```diff
@@ -105,7 +106,14 @@ object SparkSource {
   def toBaseRelation(table: FileStoreTable, _sqlContext: SQLContext): BaseRelation = {
     new BaseRelation {
       override def sqlContext: SQLContext = _sqlContext
-      override def schema: StructType = SparkTypeUtils.fromPaimonRowType(table.rowType())
+      override def schema: StructType = {
+        val schema = SparkTypeUtils.fromPaimonRowType(table.rowType())
+        if (Options.fromMap(table.options).get(SparkConnectorOptions.READ_CHANGELOG)) {
```

Comment:
`Options.fromMap(table.options).get(SparkConnectorOptions.READ_CHANGELOG)` can be extracted a util method